### PR TITLE
Move blocks to a top-level SVG while dragging

### PR DESCRIPTION
**Pull request by joshlory**
Allow blocks to be dragged outside the extents of their block space SVG.  Also ensures that the block is always at the top of the z-index while dragging.

### CODE REVIEW

File: core/ui/block_space/block_space.js [hunk 1]
```diff
@@ -328,9 +328,22 @@ Blockly.BlockSpace.prototype.createDom = function() {
...
+    Blockly.dragSvg = Blockly.createSvgElement('svg', {
+      id: 'blocklyDragCanvas',
+      width: '100%',
+      height: '100%',
+      style: 'pointer-events: none; position: absolute; top: 0; left: 0;'
+    }, document.body);
+
+    Blockly.dragCanvas = Blockly.createSvgElement('g', {'class': 'svgDragCanvas'}, Blockly.dragSvg);
+  }
+  this.svgDragCanvas_ = Blockly.dragCanvas;
```
Comment (Hamms) [inline core/ui/block_space/block_space.js]: damn, I love that this is so straightforward!

Comment (Hamms) [inline core/ui/block_space/block_space.js]: Has this been manually tested with the function editor to make sure that two blockspaces using the same drag canvas is fine?

Comment (joshlory) [inline core/ui/block_space/block_space.js]: It works correctly with the main block space / function editor block space, but I need to fix a few bugs with flyouts and mutators 😕.

Comment (Hamms) [inline core/ui/block_space/block_space.js]: blocking bugs?

Comment (joshlory) [inline core/ui/block_space/block_space.js]: Yeah, working to fix them today.

**Comment by joshlory** [comment]
I've given each drag canvas a separate `<g>` element to more closely match the setup before moving the `<g>` elements to a separate SVG.  This should take care of a few lingering issues where the offset of the drag canvas is stale when dragging in a mutator, immediately followed by dragging in the main block space.  PTAL!

**Pull request merged**